src/common.jl

```diff
@@ -16,6 +16,12 @@ const Univariate    = ArrayLikeVariate{0}
 const Multivariate  = ArrayLikeVariate{1}
 const Matrixvariate = ArrayLikeVariate{2}
 
+"""
+`F <: NamedTupleVariate{K}` specifies that the variate or a sample is of type
+`NamedTuple{K}`.
+"""
+abstract type NamedTupleVariate{K} <: VariateForm end
```

Comment:
Couldn't/Shouldn't this be a concrete subtype?
```suggestion
struct NamedTupleVariate{K} <: VariateForm end
```

Reply:
Perhaps, but this would be inconsistent with other `VariateForm` subtypes. Both `ArrayLikeVariate` and `CholeskyVariate` could be concrete subtypes but are currently defined as abstract subtypes.

Reply:
I see. I'm not sure there's a good reason for the current implementation but maybe let's be consistent for now then.

Reply:
I thought a bit more about this and I think the existing definitions might just be a bug - I've never encountered a subtype of these variate forms, and I wouldn't expect anyone to define a subtype for NamedTupleVariate either.

I also wonder if abstract type parameters might actually be worse for the compiler.

So I would suggest making it a concrete type.